### PR TITLE
Issue 1910: Upgrade our Jersey dependencies from 2.23.1 to 2.32

### LINKED ISSUE #1910: Upgrade our Jersey dependencies from 2.23.1 to 2.32
# Task Description

We need to upgrade our `jersey` dependencies from `2.23.1` to `2.32`.

# Tasks

The following tasks will need to be carried out:
* [ ] Update the version in the [strongbox-parent] project.
* [ ] Apply any required code changes, or dependency excludes.
* [ ] Test and confirm that there are no issues with the core projects ([strongbox], [strongbox-web-integration-tests], etc) after upgrading.

# Task Relationships

This task:
* It's dependent of: #1851 

# Useful Links

* [Migrating to Jersey 2.32](https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/migration.html#mig-2.26)

# Help

* Check our article on [how to upgrade dependencies and plugins].
* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @steve-todorov

[strongbox]: https://github.com/strongbox/strongbox/
[strongbox-parent]: https://github.com/strongbox/strongbox-parent/
[strongbox-web-integration-tests]: https://github.com/strongbox/strongbox-web-integration-tests/
[how to upgrade dependencies and plugins]: https://strongbox.github.io/developer-guide/upgrading-dependencies-and-plugins.html

### PR DESCRIPTION
**Pull request by ptirador**
# Pull Request Description

This pull request closes #1910

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [x] Yes, please see strongbox/strongbox-parent/pull/106
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

# Code Review And Pre-Merge Checklist

* [x] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code in hard-to-understand areas.
* [x] My changes generate no new warnings.
* [ ] I have added tests that prove my fix is effective or that my feature works.
* [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by ptirador** [comment]
@carlspring: This is ready for review.

**Comment by carlspring** [comment]
Testing in build [1911#3](https://jenkins.carlspring.org/blue/organizations/jenkins/strongbox%2Fbuilds%2Fstrongbox/detail/PR-1911/3/pipeline)

**Comment by carlspring** [comment]
The tests in Jenkins have failed. Do you ned to make the dependency upgrades together with the other pull request?

**Comment by ptirador** [comment]
Hi @carlspring, 

The problem with tests was that the `s-w-i-t` in Jenkins build were running using its `master` branch (instead of the one using the correct `strongbox-parent` version). It's been solved now.
The build [1911#4](https://jenkins.carlspring.org/blue/organizations/jenkins/strongbox%2Fbuilds%2Fstrongbox/detail/PR-1911/4/pipeline) seems to have passed ok.

File: strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/domain/GroovyScriptNames.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import java.util.Collections;
 import java.util.List;
 
-import jersey.repackaged.com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList;
```
Comment (carlspring) [inline strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/domain/GroovyScriptNames.java:8]: Thank you for spotting this and fixing it on the way! :)

File: strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/configuration/ProxyConfiguration.java [hunk 2]
```diff
@@ -6,8 +6,8 @@
 import java.util.Collections;
 import java.util.List;
 
+import com.google.common.collect.ImmutableList;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import jersey.repackaged.com.google.common.collect.ImmutableList;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/configuration/ProxyConfiguration.java:10]: Thanks! :)

**Review by carlspring — APPROVED** [review]
Thank you very much! This looks good to me! Could you please, revert the version of the parent back to `1.0-SNAPSHOT`?

Thanks! :)

**Pull request merged**